Dockerfile_mail_filter

```diff
@@ -0,0 +1,35 @@
+# https://hub.docker.com/_/debian
+FROM docker.io/debian:testing-slim AS build
+MAINTAINER Jelmer Vernooij <jelmer@jelmer.uk>
+
+# Enable forward compatibility with newer versions of Python
+ENV PYO3_USE_ABI3_FORWARD_COMPATIBILITY="1"
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update --yes \
+ && apt-get install --yes --no-install-recommends \
+        auto-apt-proxy \
+        iproute2 \
+ && apt-get upgrade --yes \
+ && apt-get satisfy --yes --no-install-recommends \
+       ## Standard packages: ./CONTRIBUTING.md
+        cargo \
+        g++ \
+        gcc \
+        libpython3-dev \
```

Comment:
mail filter is a rust binary, and meant to be called from within dovecot sieve filters. Not sure how much sense it makes to run it from docker?

Reply:
> mail filter is a rust binary, and meant to be called from within dovecot sieve filters. 

Sorry, I hadn't seen/found any docs/guide/comments saying which parts can be done this way

> Not sure how much sense it makes to run it from docker?

With the work Ive done so far, all the other commands have been in containers so far, so figured it made sense to keep everything the same.

The other item is, on the host OS, for production, we don't have rustc installed (try to keep it as light/small/minimal as possible).